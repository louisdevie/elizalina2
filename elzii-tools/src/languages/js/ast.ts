// import * as js from 'estree'

export * from 'estree'
export { BaseNodeWithoutComments as AnyNode } from 'estree'

// nodes that are not needed are commented out

/*export function arrayExpression(...elements: (js.Expression | js.SpreadElement | null)[]): js.ArrayExpression {
  return {
    type: 'ArrayExpression',
    elements
  }
}*/

/*export function arrayPattern(): js.ArrayPattern {
}*/

/*export function arrowFunctionExpression(): js.ArrowFunctionExpression {
}*/

/*export function assignmentExpression(): js.AssignmentExpression {
}*/

/*export function assignmentPattern(): js.AssignmentPattern {
}*/

/*export function awaitExpression(): js.AwaitExpression {
}*/

/*export function binaryExpression(): js.BinaryExpression {
}*/

/*export function blockStatement(): js.BlockStatement {
}*/

/*export function breakStatement(): js.BreakStatement {
}*/

/*export function callExpression(): js.CallExpression {
}*/

/*export function catchClause(): js.CatchClause {
}*/

/*export function chainExpression(): js.ChainExpression {
}*/

/*export function classBody(): js.ClassBody {
}*/

/*export function classDeclaration(): js.ClassDeclaration {
}*/

/*export function classExpression(): js.ClassExpression {
}*/

/*export function comment(): js.Comment {
}*/

/*export function conditionalExpression(): js.ConditionalExpression {
}*/

/*export function continueStatement(): js.ContinueStatement {
}*/

/*export function debuggerStatement(): js.DebuggerStatement {
}*/

/*export function doWhileStatement(): js.DoWhileStatement {
}*/

/*export function emptyStatement(): js.EmptyStatement {
}*/

/*export function exportAllDeclaration(): js.ExportAllDeclaration {
}*/

/*export function exportDefaultDeclaration(): js.ExportDefaultDeclaration {
}*/

/*export function exportNamedDeclaration(): js.ExportNamedDeclaration {
}*/

/*export function exportSpecifier(): js.ExportSpecifier {
}*/

/*export function forInStatement(): js.ForInStatement {
}*/

/*export function forOfStatement(): js.ForOfStatement {
}*/

/*export function forStatement(): js.ForStatement {
}*/

/*export function functionDeclaration(): js.FunctionDeclaration {
}*/

/*export function functionExpression(): js.FunctionExpression {
}*/

/*export function identifier(name: string): js.Identifier {
  return {
    type: 'Identifier',
    name
  }
}*/

/*export function ifStatement(): js.IfStatement {
}*/

/*export function importDeclaration(): js.ImportDeclaration {
}*/

/*export function importDefaultSpecifier(): js.ImportDefaultSpecifier {
}*/

/*export function importExpression(): js.ImportExpression {
}*/

/*export function importNamespaceSpecifier(): js.ImportNamespaceSpecifier {
}*/

/*export function importSpecifier(): js.ImportSpecifier {
}*/

/*export function labeledStatement(): js.LabeledStatement {
}*/

/*export function literal(value: any): js.Literal {
  return {
    type: 'Literal',
    value
  }
}*/

/*export function logicalExpression(): js.LogicalExpression {
}*/

/*export function memberExpression(): js.MemberExpression {
}*/

/*export function metaProperty(): js.MetaProperty {
}*/

/*export function methodDefinition(): js.MethodDefinition {
}*/

/*export function newExpression(): js.NewExpression {
}*/

/*export function objectExpression(): js.ObjectExpression {
}*/

/*export function objectPattern(): js.ObjectPattern {
}*/

/*export function privateIdentifier(): js.PrivateIdentifier {
}*/

/*export function program(): js.Program {
}*/

/*export function property(): js.Property {
}*/

/*export function propertyDefinition(): js.PropertyDefinition {
}*/

/*export function restElement(): js.RestElement {
}*/

/*export function returnStatement(): js.ReturnStatement {
}*/

/*export function sequenceExpression(): js.SequenceExpression {
}*/

/*export function spreadElement(argument: js.Expression): js.SpreadElement {
  return {
    type: 'SpreadElement',
    argument
  }
}*/

/*export function staticBlock(): js.StaticBlock {
}*/

/*export function super(): js.Super {
}*/

/*export function switchCase(): js.SwitchCase {
}*/

/*export function switchStatement(): js.SwitchStatement {
}*/

/*export function taggedTemplateExpression(): js.TaggedTemplateExpression {
}*/

/*export function templateElement(): js.TemplateElement {
}*/

/*export function templateLiteral(): js.TemplateLiteral {
}*/

/*export function thisExpression(): js.ThisExpression {
}*/

/*export function throwStatement(): js.ThrowStatement {
}*/

/*export function tryStatement(): js.TryStatement {
}*/

/*export function unaryExpression(): js.UnaryExpression {
}*/

/*export function updateExpression(): js.UpdateExpression {
}*/

/*export function variableDeclaration(): js.VariableDeclaration {
}*/

/*export function variableDeclarator(): js.VariableDeclarator {
}*/

/*export function whileStatement(): js.WhileStatement {
}*/

/*export function withStatement(): js.WithStatement {
}*/

/*export function yieldExpression(): js.YieldExpression {
}*/
