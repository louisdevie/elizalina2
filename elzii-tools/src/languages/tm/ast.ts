import { TerminalNode } from 'antlr4'
import { TMVisitor } from './visit'

export interface Node {
  accept<Result>(visitor: TMVisitor<Result>): Result
  getText(): string
}

export interface Translation extends Node {
  EOF(): TerminalNode
  NEW_LINE_list(): TerminalNode[]
  NEW_LINE(i: number): TerminalNode
  directive_list(): Directive[]
  directive(i: number): Directive
  header(): Header
  message_list(): Message[]
  message(i: number): Message
  lineComment_list(): LineComment[]
  lineComment(i: number): LineComment
  notAMessage_list(): NotAMessage[]
  notAMessage(i: number): NotAMessage
  END_OF_DIRECTIVE_list(): TerminalNode[]
  END_OF_DIRECTIVE(i: number): TerminalNode
}

export interface LineComment extends Node {
  LINE_COMMENT_START(): TerminalNode
  LINE_COMMENT_TEXT(): TerminalNode
}

export interface Header extends Node {
  HEADER_START(): TerminalNode
  EMBEDDED_CODE_CLOSING_BRACE_list(): TerminalNode[]
  EMBEDDED_CODE_CLOSING_BRACE(i: number): TerminalNode
  EMBEDDED_CODE_list(): TerminalNode[]
  EMBEDDED_CODE(i: number): TerminalNode
  EMBEDDED_CODE_OPENING_BRACE_list(): TerminalNode[]
  EMBEDDED_CODE_OPENING_BRACE(i: number): TerminalNode
}

export interface Directive extends Node {
  DIRECTIVE_NAME(): TerminalNode
  END_OF_DIRECTIVE(): TerminalNode
  DIRECTIVE_ARGUMENT_list(): TerminalNode[]
  DIRECTIVE_ARGUMENT(i: number): TerminalNode
  DIRECTIVE_ARGUMENT_SEPARATOR_list(): TerminalNode[]
  DIRECTIVE_ARGUMENT_SEPARATOR(i: number): TerminalNode
}

export interface Message extends Node {
  KEY(): TerminalNode
  SEPARATOR(): TerminalNode
  singleQuotedText(): SingleQuotedText
  doubleQuotedText(): DoubleQuotedText
}

export interface NotAMessage extends Node {
  HEADER_START(): TerminalNode
  DIRECTIVE_NAME(): TerminalNode
  NEW_LINE_list(): TerminalNode[]
  NEW_LINE(i: number): TerminalNode
  END_OF_DIRECTIVE_list(): TerminalNode[]
  END_OF_DIRECTIVE(i: number): TerminalNode
}

export interface SingleQuotedText extends Node {
  SINGLE_QUOTE(): TerminalNode
  SINGLE_QUOTE_CLOSING(): TerminalNode
  SINGLE_QUOTED_TEXT_LITERAL_list(): TerminalNode[]
  SINGLE_QUOTED_TEXT_LITERAL(i: number): TerminalNode
  SINGLE_QUOTED_ESCAPE_list(): TerminalNode[]
  SINGLE_QUOTED_ESCAPE(i: number): TerminalNode
  singleQuotedTextParameter_list(): SingleQuotedTextParameter[]
  singleQuotedTextParameter(i: number): SingleQuotedTextParameter
}

export interface DoubleQuotedText extends Node {
  DOUBLE_QUOTE(): TerminalNode
  DOUBLE_QUOTE_CLOSING(): TerminalNode
  DOUBLE_QUOTED_TEXT_LITERAL_list(): TerminalNode[]
  DOUBLE_QUOTED_TEXT_LITERAL(i: number): TerminalNode
  DOUBLE_QUOTED_ESCAPE_list(): TerminalNode[]
  DOUBLE_QUOTED_ESCAPE(i: number): TerminalNode
  doubleQuotedTextParameter_list(): DoubleQuotedTextParameter[]
  doubleQuotedTextParameter(i: number): DoubleQuotedTextParameter
}

export interface SingleQuotedTextParameter extends Node {
  SINGLE_QUOTED_PARAMETER_START(): TerminalNode
  PARAMETER_NAME(): TerminalNode
  customParameterFormat(): ShorthandParameterFormat
  parameterFormat(): ParameterFormat
  PARAMETER_END(): TerminalNode
}

export interface DoubleQuotedTextParameter extends Node {
  DOUBLE_QUOTED_PARAMETER_START(): TerminalNode
  PARAMETER_NAME(): TerminalNode
  customParameterFormat(): ShorthandParameterFormat
  parameterFormat(): ParameterFormat
  PARAMETER_END(): TerminalNode
}

export interface ParameterFormat extends Node {
  FORMAT_SEPARATOR(): TerminalNode
  EMBEDDED_CODE_CLOSING_BRACE_list(): TerminalNode[]
  EMBEDDED_CODE_CLOSING_BRACE(i: number): TerminalNode
  EMBEDDED_CODE_list(): TerminalNode[]
  EMBEDDED_CODE(i: number): TerminalNode
  EMBEDDED_CODE_OPENING_BRACE_list(): TerminalNode[]
  EMBEDDED_CODE_OPENING_BRACE(i: number): TerminalNode
}

export interface ShorthandParameterFormat extends Node {
  CUSTOM_FORMAT_SEPARATOR(): TerminalNode
  EMBEDDED_CODE_CLOSING_BRACE_list(): TerminalNode[]
  EMBEDDED_CODE_CLOSING_BRACE(i: number): TerminalNode
  EMBEDDED_CODE_list(): TerminalNode[]
  EMBEDDED_CODE(i: number): TerminalNode
  EMBEDDED_CODE_OPENING_BRACE_list(): TerminalNode[]
  EMBEDDED_CODE_OPENING_BRACE(i: number): TerminalNode
}
