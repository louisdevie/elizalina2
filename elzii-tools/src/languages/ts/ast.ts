export * from '@typescript-eslint/types/dist/generated/ast-spec'

