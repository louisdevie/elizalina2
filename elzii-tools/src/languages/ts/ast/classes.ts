import {
  AST_NODE_TYPES,
  BlockStatement,
  ClassBody,
  ClassDeclaration,
  ClassDeclarationWithName,
  ClassElement,
  MethodDefinitionNonComputedName,
  Parameter,
  TSClassImplements,
  TSInterfaceBody,
  TSInterfaceDeclaration,
  TSMethodSignature,
  TSPropertySignature,
  TSTypeAnnotation,
  TypeElement,
} from '@module/languages/ts/tsestree-spec'
import { functionExpression, FunctionOptions } from './functions'
import { identifier } from './misc'
import { TypeAliasDeclaration } from 'typescript'

export function classBody(...elements: ClassElement[]): ClassBody {
  return { type: AST_NODE_TYPES.ClassBody, body: elements }
}

export interface ClassOptions {
  abstract?: boolean
  body?: ClassBody
  implements?: TSClassImplements[]
}

export function classDeclaration(name: string, options?: ClassOptions): ClassDeclarationWithName
export function classDeclaration(name: string, options?: ClassOptions): ClassDeclaration {
  return {
    type: AST_NODE_TYPES.ClassDeclaration,
    abstract: options?.abstract ?? false,
    body: options?.body ?? classBody(),
    declare: false,
    decorators: [],
    id: identifier(name),
    implements: options?.implements ?? [],
    superClass: null,
    superTypeArguments: undefined,
    superTypeParameters: undefined,
    typeParameters: undefined,
  }
}

export interface MethodOptions extends FunctionOptions<BlockStatement> {
  kind?: 'get' | 'method' // | 'constructor' | 'set'
  // accessibility?: Accessibility | undefined
  // static?: boolean
  // override?: boolean
}

export function methodDefinition(
  name: string,
  options: MethodOptions,
): MethodDefinitionNonComputedName {
  return {
    type: AST_NODE_TYPES.MethodDefinition,
    key: identifier(name),
    computed: false,
    value: functionExpression(null, options),
    static: /* options.static ?? */ false,
    override: /* options.override ?? */ false,
    kind: options.kind ?? 'method',
    accessibility: undefined /* options.accessibility */,
    optional: false,
    decorators: [],
  }
}

export function tsClassImplements(className: string): TSClassImplements
export function tsClassImplements(className: string): TSClassImplements {
  return {
    type: AST_NODE_TYPES.TSClassImplements,
    expression: identifier(className),
    typeParameters: undefined,
    typeArguments: undefined,
  }
}

export function tsInterfaceBody(...elements: TypeElement[]): TSInterfaceBody {
  return {
    type: AST_NODE_TYPES.TSInterfaceBody,
    body: elements,
  }
}

export function tsInterfaceDeclaration(name: string): TSInterfaceDeclaration {
  return {
    type: AST_NODE_TYPES.TSInterfaceDeclaration,
    id: identifier(name),
    body: tsInterfaceBody(),
    declare: false,
    extends: [],
    typeParameters: undefined,
  }
}

export interface MethodSignatureOptions {
  kind?: 'get' | 'method' // | 'constructor' | 'set'
  // accessibility?: Accessibility | undefined
  params: Parameter[]
  returnType?: TSTypeAnnotation | undefined
}

export function tsMethodSignature(
  name: string,
  options: MethodSignatureOptions,
): TSMethodSignature {
  return {
    type: AST_NODE_TYPES.TSMethodSignature,
    key: identifier(name),
    computed: false,
    kind: options.kind ?? 'method',
    accessibility: undefined,
    optional: false,
    static: false,
    typeParameters: undefined,
    params: options.params,
    returnType: options.returnType,
    readonly: false,
  }
}
