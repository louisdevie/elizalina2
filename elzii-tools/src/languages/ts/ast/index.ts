export * from '../tsestree-spec'
export { NodeOrTokenData as AnyNode } from '../tsestree-spec'

export * from './classes'
export * from './expressions'
export * from './functions'
export * from './global'
export * from './misc'
export * from './terminals'
export * from './types'

// below are nodes that are not needed

// export function accessorProperty(): ts.AccessorProperty {}
// export function arrayPattern(): ts.ArrayPattern {}
// export function assignmentPattern(): ts.AssignmentPattern {}
// export function awaitExpression(): ts.AwaitExpression {}
// export function binaryExpression(): ts.BinaryExpression {}
// export function breakStatement(): ts.BreakStatement {}
// export function catchClause(): ts.CatchClause {}
// export function chainExpression(): ts.ChainExpression {}
// export function classExpression(): ts.ClassExpression {}
// export function conditionalExpression(): ts.ConditionalExpression {}
// export function continueStatement(): ts.ContinueStatement {}
// export function debuggerStatement(): ts.DebuggerStatement {}
// export function decorator(): ts.Decorator {}
// export function doWhileStatement(): ts.DoWhileStatement {}
// export function emptyStatement(): ts.EmptyStatement {}
// export function exportAllDeclaration(): ts.ExportAllDeclaration {}
// export function exportSpecifier(): ts.ExportSpecifier {}
// export function forInStatement(): ts.ForInStatement {}
// export function forOfStatement(): ts.ForOfStatement {}
// export function forStatement(): ts.ForStatement {}
// export function functionDeclaration(): ts.FunctionDeclaration {}
// export function ifStatement(): ts.IfStatement {}
// export function importAttribute(): ts.ImportAttribute {}
// export function importNamespaceSpecifier(): ts.ImportNamespaceSpecifier {}
// export function jSXAttribute(): ts.JSXAttribute {}
// export function jSXClosingElement(): ts.JSXClosingElement {}
// export function jSXClosingFragment(): ts.JSXClosingFragment {}
// export function jSXElement(): ts.JSXElement {}
// export function jSXEmptyExpression(): ts.JSXEmptyExpression {}
// export function jSXExpressionContainer(): ts.JSXExpressionContainer {}
// export function jSXFragment(): ts.JSXFragment {}
// export function jSXIdentifier(): ts.JSXIdentifier {}
// export function jSXMemberExpression(): ts.JSXMemberExpression {}
// export function jSXNamespacedName(): ts.JSXNamespacedName {}
// export function jSXOpeningElement(): ts.JSXOpeningElement {}
// export function jSXOpeningFragment(): ts.JSXOpeningFragment {}
// export function jSXSpreadAttribute(): ts.JSXSpreadAttribute {}
// export function jSXSpreadChild(): ts.JSXSpreadChild {}
// export function jSXText(): ts.JSXText {}
// export function labeledStatement(): ts.LabeledStatement {}
// export function logicalExpression(): ts.LogicalExpression {}
// export function metaProperty(): ts.MetaProperty {}
// export function objectPattern(): ts.ObjectPattern {}
// export function privateIdentifier(): ts.PrivateIdentifier {}
// export function restElement(): ts.RestElement {}
// export function sequenceExpression(): ts.SequenceExpression {}
// export function spreadElement(): ts.SpreadElement {}
// export function staticBlock(): ts.StaticBlock {}
// export function super(): ts.Super {}
// export function switchCase(): ts.SwitchCase {}
// export function switchStatement(): ts.SwitchStatement {}
// export function taggedTemplateExpression(): ts.TaggedTemplateExpression {}
// export function throwStatement(): ts.ThrowStatement {}
// export function tryStatement(): ts.TryStatement {}
// export function unaryExpression(): ts.UnaryExpression {}
// export function updateExpression(): ts.UpdateExpression {}
// export function whileStatement(): ts.WhileStatement {}
// export function withStatement(): ts.WithStatement {}
// export function yieldExpression(): ts.YieldExpression {}
// export function tsAbstractAccessorProperty(): ts.TSAbstractAccessorProperty {}
// export function tsAbstractKeyword(): ts.TSAbstractKeyword {}
// export function tsAbstractMethodDefinition(): ts.TSAbstractMethodDefinition {}
// export function tsAbstractPropertyDefinition(): ts.TSAbstractPropertyDefinition {}
// export function tsAsExpression(): ts.TSAsExpression {}
// export function tsAsyncKeyword(): ts.TSAsyncKeyword {}
// export function tsBigIntKeyword(): ts.TSBigIntKeyword {}
// export function tsBooleanKeyword(): ts.TSBooleanKeyword {}
// export function tsCallSignatureDeclaration(): ts.TSCallSignatureDeclaration {}
// export function tsConditionalType(): ts.TSConditionalType {}
// export function tsConstructorType(): ts.TSConstructorType {}
// export function tsConstructSignatureDeclaration(): ts.TSConstructSignatureDeclaration {}
// export function tsDeclareFunction(): ts.TSDeclareFunction {}
// export function tsDeclareKeyword(): ts.TSDeclareKeyword {}
// export function tsEmptyBodyFunctionExpression(): ts.TSEmptyBodyFunctionExpression {}
// export function tsEnumDeclaration(): ts.TSEnumDeclaration {}
// export function tsEnumMember(): ts.TSEnumMember {}
// export function tsExportAssignment(): ts.TSExportAssignment {}
// export function tsExportKeyword(): ts.TSExportKeyword {}
// export function tsExternalModuleReference(): ts.TSExternalModuleReference {}
// export function tsFunctionType(): ts.TSFunctionType {}
// export function tsInstantiationExpression(): ts.TSInstantiationExpression {}
// export function tsImportEqualsDeclaration(): ts.TSImportEqualsDeclaration {}
// export function tsImportType(): ts.TSImportType {}
// export function tsIndexedAccessType(): ts.TSIndexedAccessType {}
// export function tsIndexSignature(): ts.TSIndexSignature {}
// export function tsInferType(): ts.TSInferType {}
// export function tsInterfaceHeritage(): ts.TSInterfaceHeritage {}
// export function tsIntersectionType(): ts.TSIntersectionType {}
// export function tsIntrinsicKeyword(): ts.TSIntrinsicKeyword {}
// export function tsLiteralType(): ts.TSLiteralType {}
// export function tsMappedType(): ts.TSMappedType {}
// export function tsModuleBlock(): ts.TSModuleBlock {}
// export function tsModuleDeclaration(): ts.TSModuleDeclaration {}
// export function tsNamedTupleMember(): ts.TSNamedTupleMember {}
// export function tsNamespaceExportDeclaration(): ts.TSNamespaceExportDeclaration {}
// export function tsNeverKeyword(): ts.TSNeverKeyword {}
// export function tsNonNullExpression(): ts.TSNonNullExpression {}
// export function tsNullKeyword(): ts.TSNullKeyword {}
// export function tsObjectKeyword(): ts.TSObjectKeyword {}
// export function tsOptionalType(): ts.TSOptionalType {}
// export function tsParameterProperty(): ts.TSParameterProperty {}
// export function tsPrivateKeyword(): ts.TSPrivateKeyword {}
// export function tsPropertySignature(): ts.TSPropertySignature {}
// export function tsProtectedKeyword(): ts.TSProtectedKeyword {}
// export function tsPublicKeyword(): ts.TSPublicKeyword {}
// export function tsQualifiedName(): ts.TSQualifiedName {}
// export function tsReadonlyKeyword(): ts.TSReadonlyKeyword {}
// export function tsRestType(): ts.TSRestType {}
// export function tsSatisfiesExpression(): ts.TSSatisfiesExpression {}
// export function tsStaticKeyword(): ts.TSStaticKeyword {}
// export function tsSymbolKeyword(): ts.TSSymbolKeyword {}
// export function tsTemplateLiteralType(): ts.TSTemplateLiteralType {}
// export function tsThisType(): ts.TSThisType {}
// export function tsTupleType(): ts.TSTupleType {}
// export function tsTypeAliasDeclaration(): ts.TSTypeAliasDeclaration {}
// export function tsTypeAssertion(): ts.TSTypeAssertion {}
// export function tsTypeLiteral(): ts.TSTypeLiteral {}
// export function tsTypeOperator(): ts.TSTypeOperator {}
// export function tsTypeParameter(): ts.TSTypeParameter {}
// export function tsTypeParameterDeclaration(): ts.TSTypeParameterDeclaration {}
// export function tsTypePredicate(): ts.TSTypePredicate {}
// export function tsTypeQuery(): ts.TSTypeQuery {}
// export function tsUndefinedKeyword(): ts.TSUndefinedKeyword {}
// export function tsUnionType(): ts.TSUnionType {}
// export function tsUnknownKeyword(): ts.TSUnknownKeyword {}
// export function tsVoidKeyword(): ts.TSVoidKeyword {}
