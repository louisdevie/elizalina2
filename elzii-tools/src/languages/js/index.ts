import { AnyNode } from './ast'
import { JSPrinterImpl } from './print'

export { JSVisitor } from './visit'

export interface JSPrinter {
  print(code: AnyNode): Promise<string>
}

export function getJSPrinter(): JSPrinter {
  return new JSPrinterImpl()
}
