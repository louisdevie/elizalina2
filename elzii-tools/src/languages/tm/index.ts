import { AntlrTMParserAdapter } from './parse'
import { Translation as TMTranslation } from './ast'

export interface TMParser {
  parse(text: string): Promise<TMTranslation>
}

export const defaultTMParser: TMParser = new AntlrTMParserAdapter()
