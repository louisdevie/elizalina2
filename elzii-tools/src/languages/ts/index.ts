import { CodeStyle } from '@module/options'
import { NodeOrTokenData } from './ast'

export { default as ProgramBuilder } from './ProgramBuilder'

export interface TSPrinter {
  print(code: NodeOrTokenData): Promise<string>
}

export function getTSPrinter(codeStyle: CodeStyle): TSPrinter {
  return {}
}
