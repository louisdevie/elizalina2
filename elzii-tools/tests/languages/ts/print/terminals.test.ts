import { getTSPrinter } from '@module/languages/ts'
import * as ts from '@module/languages/ts/ast'

test('printing a BlockComment', async () => {
  const printer = getTSPrinter()

  const singleLine = ts.blockComment('Single line')
  expect(await printer.print(singleLine)).toMatchSnapshot()

  const multiLine = ts.blockComment('Multi\nline')
  expect(await printer.print(multiLine)).toMatchSnapshot()
})

test('printing a Literal', async () => {
  const printer = getTSPrinter()

  const stringLiteral = ts.literal('Hello chat')
  expect(await printer.print(stringLiteral)).toMatchSnapshot()
})
