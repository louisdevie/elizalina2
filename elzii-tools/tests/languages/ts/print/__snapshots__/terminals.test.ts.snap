// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`printing a BlockComment 1`] = `"/* Single line */"`;

exports[`printing a BlockComment 2`] = `
"/* Multi
 * line
 */"
`;

exports[`printing a Literal 1`] = `"'Hello chat'"`;
